feat: experimental payment v2 migration script (#616)
omisego/plasma-contracts: plasma_framework/docs/deploying/deploying.md

## Migration scripts
Truffle migration scripts run in the order of the numbered prefix of the filename. The naming convention is as follows:

- 1-98: Initial setup and main contracts like PlasmaFramework, SpendingConditionRegistry, etc.
- 99: Renounce the SpendingConditionRegistry because this has to happen before any ExitGames
- 100: First ExitGame contracts
- 200: Second ExitGame contracts
- 300: payment V2 experiment contracts
- 1000: Cleanup, dev/test contracts and output
- 2000: Future extensions, new exit games, etc.

## ENV VAR settings

> - TENDERLY: set to `true` when needed to push the contract to tenderly. Default not pushing to tenderly.

Reviewer: ```suggestion
- TENDERLY: set to `true` when you need to push the contract to tenderly. Default not pushing to tenderly.
```

> - TENDERLY: set to `true` when needed to push the contract to tenderly. Default not pushing to tenderly.

Author: > Default: `false`.

This is slightly incorrect. It is default to `undefined`

> - TENDERLY: set to `true` when needed to push the contract to tenderly. Default not pushing to tenderly.

Reviewer 2: actually it is false :
```const isExperiment = process.env.EXPERIMENT || false;```

> - TENDERLY: set to `true` when needed to push the contract to tenderly. Default not pushing to tenderly.

Author: eh..yes. the `|| false` part make `isExperiment` false. I think what I want to point out is that the env var itself (`process.env.EXPERIMENT`) would be considered `undefined`, so it is slightly incorrect in this sense.

> - DEPLOY_TEST_CONTRACTS: set to `true` when need to deploy some testing contracts like mock, wrapper for conformance testing. Default not deploying test contracts.

Reviewer: ```suggestion
- DEPLOY_TEST_CONTRACTS: set to `true` when you need to deploy some testing contracts like mock, wrapper for conformance testing. By default test contracts are not deployed.
```

> - EXPERIMENT: set to `true` when needed to deploy experimental contracts. For instance, deploying new payment exit game contract to test the flow of extension.

Reviewer: ```suggestion
- EXPERIMENT: set to `true` when you need to deploy experimental contracts. For instance, deploy a new payment exit game contract to test the flow to extend the system.
```

> - MULTI_EXIT_GAME_EXPERIMENT: This is the feature flag for the experiment of doing a multiple exit games. This experiment focuses on extending to a payment v2 feature that is a clone of payment v1. So it would be deploying the same contract with slightly different parameters on indicating the tx type. set to `true` when you need to deploy experimental contracts.

Reviewer 2: ```suggestion
- MULTI_EXIT_GAME_EXPERIMENT: This is the feature flag for the experiment of doing a multiple exit games. This experiment focuses on extending to a payment v2 feature that is a clone of payment v1. So it would be deploying the same contract with slightly different parameters on indicating the tx type. Set to `true` when you need to deploy experimental contracts.
```
